packages/wrangler/src/dev/miniflare.ts

```diff
@@ -355,8 +394,88 @@ export function buildMiniflareBindingOptions(config: MiniflareBindingsConfig): {
 				.join("\n"),
 	};
 
+	// Setup service bindings to external services
+	const serviceBindings: NonNullable<WorkerOptions["serviceBindings"]> = {
+		...config.serviceBindings,
+	};
+	for (const service of config.services ?? []) {
+		if (service.service === config.name) {
+			// If this is a service binding to the current worker, don't bother using
+			// the dev registry to look up the address, just bind to it directly.
+			serviceBindings[service.binding] = getName(config);
+			continue;
+		}
+
+		const target = config.workerDefinitions?.[service.service];
+		if (target?.host === undefined || target.port === undefined) {
+			// If the target isn't in the registry, always return an error response
+			// TODO(someday): could we do some sort of `Proxy`-prototype thing here
+			//  for a nice error message if a user tried to call an RPC method?
+			serviceBindings[service.binding] = () => {
+				const message = `[wrangler] Couldn't find \`wrangler dev\` session for service "${service.service}" to proxy to`;
+				return new Response(message, { status: 503 });
+			};
+		} else {
+			// Otherwise, try to build an `external` service to it. `external`
+			// services support JSRPC over HTTP CONNECT using a special hostname.
+			// Refer to https://github.com/cloudflare/workerd/pull/1757 for details.
+			let address: `${string}:${number}`;
+			if (service.entrypoint !== undefined) {
+				// If the user has requested a named entrypoint...
+				if (target.entrypointAddresses === undefined) {
+					// ...but the "server" `wrangler` hasn't provided any because it's too
+					// old, throw.
+					throw new UserError(
+						`The \`wrangler dev\` session for service "${service.service}" does not support proxying entrypoints. Please upgrade "${service.service}"'s \`wrangler\` version.`
+					);
+				}
+				const entrypointAddress =
+					target.entrypointAddresses[service.entrypoint];
+				if (entrypointAddress === undefined) {
+					// ...but the named entrypoint doesn't exist, throw
+					throw new UserError(
+						`The \`wrangler dev\` session for service "${service.service}" does export an entrypoint named "${service.entrypoint}"`
+					);
+				}
+				address = `${entrypointAddress.host}:${entrypointAddress.port}`;
+			} else {
+				// Otherwise, if the user hasn't specified a named entrypoint, assume
+				// they meant to bind to the `default` entrypoint.
+				const defaultEntrypointAddress =
+					target.entrypointAddresses?.["default"];
+				if (defaultEntrypointAddress === undefined) {
+					// If the "server" `wrangler` is too old to provide direct entrypoint
+					// addresses, fallback to sending requests directly to the target...
+					if (target.protocol === "https") {
+						// ...unless the target is listening on HTTPS, in which case throw.
```

Comment:
Is this a limitation of local dev service bindings today? Can I not bind to a service if that server is set up to use HTTPS?